cmd/argoexec/commands/wait.go

```diff
@@ -32,20 +30,14 @@ func waitContainer(ctx context.Context) error {
 	defer stats.LogStats()
 	stats.StartStatsTicker(5 * time.Minute)
 
-	// use a function to constrain the scope of ctx
-	func() {
-		// this allows us to gracefully shutdown, capturing artifacts
-		ctx, cancel := signal.NotifyContext(ctx, syscall.SIGTERM)
-		defer cancel()
```

Comment:
@sxllwx I think you're suggesting:

1. `NotifyContext` catches a SIGTERM (probably sent from the controller).
2. This block will exit.
3. `stop()` will be invoked, clearing the signal handler.
4. Another SIGTERM occurs (controller again), but without a handler we get default behaviour.
5. Default behaviour is to exit (does this mean crash?).

If that is what you think is correct, then my fix should work.

Reply:
Yes. I tried to look at the default processing method of Golang's runtime, and the relevant code:

https://github.com/golang/go/blob/master/src/runtime/signal_unix.go#L1068-L1074

https://github.com/golang/go/blob/master/src/runtime/signal_unix.go#L869-L875

Here's a test workflow I've been using for the past weekend, available for your use.

```yaml
apiVersion: argoproj.io/v1alpha1
kind: CronWorkflow
metadata:
   generateName: etcd-code-artifact-
spec:
   schedule: "*/1 * * * *"
   concurrencyPolicy: "Forbid"
   startingDeadlineSeconds: 0
   workflowSpec:
    entrypoint: run
    imagePullSecrets:
    - name: ccr-secret
    templates:
      - name: run
        inputs:
            artifacts:
            - name: repo-artifact
              path: /src
              git:
                repo: "https://github.com/etcd-io/etcd.git"
        container:
          image: "python:3.9-buster"
          command: [ sh, -c ]
          args: [ "git status && ls && pwd" ]
          workingDir: /src
        outputs:
          artifacts:
            - name: repo-artifact
              path: /src
```